components/search/src/types.rs

```diff
@@ -0,0 +1,197 @@
+    /// is included in the base.
+    pub search_term_param_name: Option<String>,
+}
+
+/// The URLs associated with the search engine.
+#[derive(Debug, uniffi::Record)]
+pub struct SearchEngineUrls {
+    /// The URL to use for searches.
+    pub search: SearchEngineUrl,
+
+    /// The URL to use for suggestions.
+    pub suggestions: Option<SearchEngineUrl>,
+
+    /// The URL to use for trending suggestions.
+    pub trending: Option<SearchEngineUrl>,
+}
+
+/// A definition for an individual search engine to be presented to the user.
+#[derive(Debug, uniffi::Record)]
+pub struct SearchEngineDefinition {
+    /// An optional list of aliases for this engine.
+    pub aliases: Option<Vec<String>>,
+
+    /// The classification of search engine according to the main search types
+    /// (e.g. general, shopping, travel, dictionary). Currently, only marking as
+    /// a general search engine is supported.
+    /// On Android, only general search engines may be selected as "default"
+    /// search engines.
+    pub classification: String,
+
+    /// The identifier of the search engine. This is used as an internal
+    /// identifier, e.g. for saving the user's settings for the engine. It is
+    /// also used to form the base telemetry id and may be extended by telemetrySuffix.
+    pub identifier: String,
+
+    /// The user visible name of the search engine.
+    pub name: String,
+
+    /// The partner code for the engine. This will be inserted into parameters
+    /// which include `{partnerCode}`.
+    pub partner_code: Option<String>,
+
+    /// Optional suffix that is appended to the search engine identifier
+    /// following a dash, i.e. `<identifier>-<suffix>`
+    pub telemetry_suffix: Option<String>,
+
+    /// The URLs associated with the search engine.
+    pub urls: SearchEngineUrls,
+
+    /// A hint to the order that this engine should be in the engine list. This
+    /// is derived from the `engineOrders` section of the search configuration.
+    /// The higher the number, the nearer to the front it should be.
+    /// If the number is not specified, other methods of sorting may be relied
+    /// upon (e.g. alphabetical).
+    pub order_hint: Option<u8>,
+}
+
+/// Details of the search engines to display to the user, generated as a result
+/// of processing the search configuration.
+#[derive(Debug, uniffi::Record)]
+pub struct RefinedConfig {
+    /// A sorted list of engines. Clients may use the engine in the order that
+    /// this list is specified, or they may implement their own order if they
+    /// have other requirements.
+    ///
+    /// The application default engines should not be assumed from this order in
+    /// case of future changes.
+    ///
+    /// The sort order is:
+    ///
+    /// * Application Default Engine
+    /// * Application Default Engine for Private Mode (if specified & different)
+    /// * Engines sorted by descending `orderHint`
```

Comment:
I believe we deprecated `orderHint`, and `engineOrders` is now used for the same purpose. To be clear,  in the code we still use `orderHint` but it's really from the ordering of `engineOrders`. 

I remember we decided to not add `orderHint` property in the new config. I'm wondering if it would be clearer to reference `engineOrders` directly instead of `orderHint`, since the `orderHint` property does not exist in search-config-v2? Thoughts?

Reply:
Note that I've updated the comment to reference `SearchEngineDefinition.orderHint`, which should hopefully make this a bit clearer.

Reply:
LGTM :)